gym/envs/mujoco/mujoco_env.py

```diff
@@ -136,9 +173,21 @@ def do_simulation(self, ctrl, n_frames):
         if np.array(ctrl).shape != self.action_space.shape:
             raise ValueError("Action dimension mismatch")
 
-        self.sim.data.ctrl[:] = ctrl
+        if self._mujoco_bindings.__name__ == "mujoco_py":
+            self.sim.data.ctrl[:] = ctrl
+        else:
+            self.data.ctrl[:] = ctrl
         for _ in range(n_frames):
-            self.sim.step()
+            if self._mujoco_bindings.__name__ == "mujoco_py":
+                self.sim.step()
+            else:
+                self._mujoco_bindings.mj_step(self.model, self.data)
+
+        # As of MuJoCo 2.0, force-related quantities like cacc are not computed
+        # unless there's a force sensor in the model.
+        # See https://github.com/openai/gym/issues/1541
+        if self._mujoco_bindings.__name__ != "mujoco_py":
+            self._mujoco_bindings.mj_rnePostConstraint(self.model, self.data)
```

Comment:
This is not commented as in https://github.com/openai/gym/pull/2595/commits/ab380345dc67e5c8b767bd2f2dcea5722161c412, which causes regression in Ant-v4. See https://wandb.ai/costa-huang/cleanRL/reports/-4-19-MuJoCo-v4-vs-v2-CleanRL-s-PPO--VmlldzoxODYzODM0, https://github.com/vwxyzjn/validate-new-gym-mujoco-envs

![5AF41CE3-F3EB-4040-99CF-3FC722F30565](https://user-images.githubusercontent.com/5555347/164099712-80d9cb0d-b0ad-477f-a41d-f76a07b911eb.jpeg)

Reply:
Computing wrong quantities on purpose to avoid regression of performance makes no sense to me. It seems way more appropriate to just remove the contact forces from the observation space and reward computation, instead of carrying out computations pretending it is doing what it is supposed to do.

Reply:
I agree with @duburcqa
Doesn't this change make the benchmark compatible with MuJoCo < 2.0? Or do all those three (mujoco-py with MuJoCo < 2.0, mujoco-py with MuJoCo >= 2.0, and mujoco) produce different results?

Reply:
Theoretically, for this change in particular it is the same as mujoco < 2.0, but there are probably other internal changes in mujoco binary so I doubt the result would be exactly the same as before.